### ISSUE TITLE
Crash on IDEA 15 

### ISSUE THREAD

**Issue by JacobDorman** [ishchenko/idea-nginx#27]
**EAP IU-143.379.11**

```
Plugin 'ideanginx9' failed to initialize and will be disabled.  Please restart IntelliJ IDEA.

java.lang.NoSuchMethodError: com.intellij.openapi.components.ex.ComponentManagerEx.registerComponent(Lcom/intellij/openapi/components/ComponentConfig;Lcom/intellij/openapi/extensions/PluginDescriptor;)V
    at net.ishchenko.idea.nginx.platform.PlatformDependentToolsInstaller.<init>(PlatformDependentToolsInstaller.java:56)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
    at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
    at org.picocontainer.defaults.ConstructorInjectionComponentAdapter$1.run(ConstructorInjectionComponentAdapter.java:220)
    at org.picocontainer.defaults.ThreadLocalCyclicDependencyGuard.observe(ThreadLocalCyclicDependencyGuard.java:53)
    at org.picocontainer.defaults.ConstructorInjectionComponentAdapter.getComponentInstance(ConstructorInjectionComponentAdapter.java:248)
    at com.intellij.util.pico.ConstructorInjectionComponentAdapter.getComponentInstance(ConstructorInjectionComponentAdapter.java:59)
    at com.intellij.openapi.components.impl.ComponentManagerImpl$ComponentConfigComponentAdapter.getComponentInstance(ComponentManagerImpl.java:495)
    at com.intellij.openapi.components.impl.ComponentManagerImpl.createComponents(ComponentManagerImpl.java:123)
    at com.intellij.openapi.application.impl.ApplicationImpl.access$801(ApplicationImpl.java:90)
    at com.intellij.openapi.application.impl.ApplicationImpl$12.run(ApplicationImpl.java:496)
    at com.intellij.openapi.progress.impl.CoreProgressManager$2.run(CoreProgressManager.java:142)
    at com.intellij.openapi.progress.impl.CoreProgressManager.a(CoreProgressManager.java:446)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:392)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:54)
    at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:127)
    at com.intellij.openapi.application.impl.ApplicationImpl.createComponents(ApplicationImpl.java:505)
    at com.intellij.openapi.components.impl.ComponentManagerImpl.init(ComponentManagerImpl.java:107)
    at com.intellij.openapi.application.impl.ApplicationImpl.load(ApplicationImpl.java:454)
    at com.intellij.openapi.application.impl.ApplicationImpl.load(ApplicationImpl.java:446)
    at com.intellij.idea.IdeaApplication.run(IdeaApplication.java:194)
    at com.intellij.idea.MainImpl$1$1$1.run(MainImpl.java:64)
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:749)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:702)
    at java.awt.EventQueue$3.run(EventQueue.java:696)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:719)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:360)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by bungle** [comment]
Same happens with PhpStorm 10, 9.x works fine.

**Comment by bungle** [comment]
Seems to be related with #26.

**Comment by elpado** [comment]
Can confirm that the version do not work with PhpStorm 10.

**Comment by cmosguy** [comment]
Anyone going to fix this?

**Comment by albabar** [comment]
Same here with RubyMine 8.

**Comment by metyl** [comment]
The problem was fixed by @develar in #25 but the binary versions was 
**not** uploaded to plugins repository, as a workaround You can install plugin provided in PR.

**Comment by ishchenko** [comment]
Reuploaded. Check please

**Comment by mxxcon** [comment]
Something strange is happening. I see there's plugin version 0.1.0, when I try to install it and restart PyCharm, after restart this plugin is missing and listed as installable again. If I try to install it again, after restart of PyCharm it's missing again. :confused:

**Comment by cmosguy** [comment]
@mxxcon @ishchenko The same thing is happenign for me when I install the 0.1.0 version.  When I restart PHPStorm it does not list the plugin as installed or activated.  Can you please fix again and check that it works for you, then upload it?

Thanks!

**Comment by mxxcon** [comment]
PyCharm CE also experience missing plugin problem.

**Comment by bungle** [comment]
Missing plugins are probably because IDE will mark the plugin crashing and doesn't try to load it again on next start.

**Comment by mxxcon** [comment]
But it's as if it doesn't get installed at all..After each prompted restart I can go to `settings / plugins / browse repos` and it's listed there with "Install" button.

**Comment by mohamnag** [comment]
I'm using IntelliJ 15 and on Mac OS X.

I never had this plugin installed before, today I tried to install it. No matter how I try to do it, when I select it and install it, after restart the plugin is not installed and can be installed again. looking into plugins directory (`~/Library/Application\ Support/IntelliJIdea15`) there is a `lib` directory which obviously contains the files from this plugin but they are ignored.

**Comment by tareqy** [comment]
Same Here, i have PHPStorm 10 / PyCharm 5.0.1, i've installed already 10 times, restarted the IDE, and the plguin is not installed.

**Comment by mohamnag** [comment]
I think there is a problem with current binary published to repo, see here #25

**Comment by agileago** [comment]
+1 same happened on webstorm

**Comment by amittapalli** [comment]
Has this issue been resolved?  I am referring to the problem with the current/fixed binary in https://github.com/ishchenko/idea-nginx/pull/25.
The last comment above on Dec 19 indicated that the same happened with web storm.

**Comment by luxp** [comment]
Still has the same question

**Comment by bungle** [comment]
There doesn't seem to be any developer in this project anymore (?), so it might be a good time for someone to fork this.

**Comment by cmosguy** [comment]
I agree the developer has absconded .

**Comment by ishchenko** [comment]
Should be closed in #41
